### PR TITLE
[AbuseCH] Empty Strings Causing False Positives with Threat Intel Hash Indicator Match Rule

### LINKED ISSUE #7101: [AbuseCH] Empty String Values for Hashes Causing False Positive Detection Rule Alerts
## Describe the Issue

Some AbuseCH documents are ingested with empty strings for hash values. This is causing an infinite number of detection rule hits and effectively rendering the rule unusable. 

## Related Issues

[detection-rules/#2954](https://github.com/elastic/detection-rules/issues/2954)

## Sample Data

```
{
    "input": {
      "type": "httpjson"
    },
    "agent": {
      "name": "Redacted",
      "id": "cdcd8072-1d30-414a-aa71-c96d2cb5e259",
      "type": "filebeat",
      "ephemeral_id": "e58cd5a2-1d47-4b89-b490-c1e13b364d7f",
      "version": "8.8.1"
    },
    "@timestamp": "2023-06-22T07:35:41.076Z",
    "ecs": {
      "version": "8.8.0"
    },
    "related": {
      "hash": [
        "B9DAFC265A8BFEAA7EB30A2BA415310235381E48",
        "8E45F8B7642FA964FE628039040D5D77A7D3F177687A6969ECD000C83826EA4C94E3F6B615A5091DA0CDBFE86D183CCE",
        "7CC8EF889A24D8BE46158ED9525EDB3EFE4B872709EDFE4C565FB562271969EE",
        "T102C59EE8D06B40D2FC076EC468287AD7073231B3DEE50438276E7A089F7BDA95549E5E",
        "24576:JILK2NYDAYN0O6PUCD65X8HVLTFBQICQT/UC/FVWG/X5YIEYBF1KMXMWLKRA:",
        "67274E089D17FCBDB0A31877D7155622",
        ""
      ]
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "ti_abusech.malwarebazaar"
    },
    "abusech": {
      "malwarebazaar": {
        "intelligence": {
          "downloads": 109,
          "uploads": 1
        },
        "tags": [
          "exe"
        ],
        "anonymous": 0,
        "code_sign": []
      }
    },
    "elastic_agent": {
      "id": "cdcd8072-1d30-414a-aa71-c96d2cb5e259",
      "version": "8.8.1",
      "snapshot": false
    },
    "threat": {
      "indicator": {
        "geo": {
          "country_iso_code": "NL"
        },
        "first_seen": "2023-06-22T07:31:33.000Z",
        "file": {
          "extension": "exe",
          "size": 2740224,
          "mime_type": "application/x-dosexec",
          "pe": {
            "imphash": ""
          },
          "name": "67274e089d17fcbdb0a31877d7155622.exe",
          "hash": {
            "sha1": "b9dafc265a8bfeaa7eb30a2ba415310235381e48",
            "sha384": "8e45f8b7642fa964fe628039040d5d77a7d3f177687a6969ecd000c83826ea4c94e3f6b615a5091da0cdbfe86d183cce",
            "sha256": "7cc8ef889a24d8be46158ed9525edb3efe4b872709edfe4c565fb562271969ee",
            "tlsh": "T102C59EE8D06B40D2FC076EC468287AD7073231B3DEE50438276E7A089F7BDA95549E5E",
            "ssdeep": "24576:jIlK2NYDaYn0o6PuCD65x8HvLtFBqicQT/UC/FVWg/X5YIEYBF1kmXmWLKRa:",
            "md5": "67274e089d17fcbdb0a31877d7155622"
          },
          "elf": {}
        },
        "provider": "abuse_ch",
        "type": "file"
      },
      "software": {}
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2023-06-22T07:35:43Z",
      "created": "2023-06-22T07:35:41.076Z",
      "kind": "enrichment",
      "category": "threat",
      "type": "indicator",
      "dataset": "ti_abusech.malwarebazaar"
    },
    "tags": [
      "forwarded",
      "abusech-malwarebazaar"
    ]
  }
```

### PR DESCRIPTION
**Pull request by MakoWish**
## Type of change
- Bug

## What does this PR do?

This PR adds a script processor to remove fields with empty string values to resolve false-positives with the `Threat Intel Hash Indicator Match` Detection Rule. 

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have incremented the version in my package's `manifest.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #7101 
- Relates https://github.com/elastic/detection-rules/issues/2954

### CODE REVIEW

File: packages/ti_abusech/data_stream/malware/_dev/test/pipeline/test-malware-ndjson.log-expected.json [hunk 1]
```diff
@@ -1141,6 +1140,47 @@
                     "type": "file"
                 }
             }
+        },
+        {
+            "abusech": {
+                "malware": {}
```
Comment (efd6) [inline packages/ti_abusech/data_stream/malware/_dev/test/pipeline/test-malware-ndjson.log-expected.json:1146]: Why is this not removed?

Comment (MakoWish) [reply packages/ti_abusech/data_stream/malware/_dev/test/pipeline/test-malware-ndjson.log-expected.json:1146]: I just moved the new script to be the last processor. There was a remove processor as the final step, and it removed `abusech.malware.firstseen`. That was leaving the empty hash. I will push the new changes shortly.

**Review by efd6 — APPROVED** [review]

**Pull request merged**